### PR TITLE
Adding B-Tree

### LINKED ISSUE #175: Adding B-Tree data structure
#### References/Other comments
https://en.wikipedia.org/wiki/B-tree

### PR DESCRIPTION
**Pull request by Arvind-raj06**
#### References to other Issues or PRs or Relevant literature
Fixes #175. See
https://github.com/codezonediitj/pydatastructs/issues/175

#### Brief description of what is fixed or changed
This commit is the basic implementation of B-Tree and it's Node. 

#### Other comments
The code works just fine and this is also counted under SWOC. If you have any query ping me

### CODE REVIEW

**Comment by Arvind-raj06** [comment]
@czgdp1807 I'm working on the coverage will make a commit for that soon.

File: pydatastructs/trees/btree.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    ========
+
+    >>> from pydatastructs.trees.btree import BTree
+    >>> bt = BTree(5)
+    >>> bt.insert(1)
+    >>> bt.insert(2)
+    >>> bt.insert(3)
+    >>> bt.insert(4)
+    >>> list(bt)
+    [1, 2, 3, 4]
```
Comment (sidhu1012) [inline pydatastructs/trees/btree.py:29]: BTree is a self balancing tree so input should be arranged accordingly and printing Tree should give be of form of any order(Post/Pre/In)
(Read more here)[https://www.geeksforgeeks.org/introduction-of-b-tree-2/]

Comment (Arvind-raj06) [reply pydatastructs/trees/btree.py:29]: I'm not getting your point it's just example for users to see

Comment (Arvind-raj06) [reply pydatastructs/trees/btree.py:29]: For you reference it's already in order traversal as per the reference https://www.geeksforgeeks.org/introduction-of-b-tree-2/

**Comment by czgdp1807** [comment]
B-Tree is an External Memory data structure or at least it was developed for effectively managing data on disk. So, adding B-Tree is a great idea but it should wait until we are done with the goals for first version.

**Comment by Arvind-raj06** [comment]
> B-Tree is an External Memory data structure or at least it was developed for effectively managing data on disk. So, adding B-Tree is a great idea but it should wait until we are done with the goals for first version.

I thought It will help in milestone like 2-3-4 tree as it's simple just call B-tree with 4 order and b-tree already in [v.0.0.1](https://github.com/codezonediitj/pydatastructs/milestone/1)
 [B-Tree](https://github.com/codezonediitj/pydatastructs/issues/175)

**Comment by Arvind-raj06** [comment]
@czgdp1807 Pls reply

**Comment by czgdp1807** [comment]
You may add, (a, b)-Tree but B-Tree isn't something which supposed stay in memory at all times.

**Comment by Arvind-raj06** [comment]
> You may add, (a, b)-Tree but B-Tree isn't something which supposed stay in memory at all times.

Ok I thought as it's in milestone it must be added.

**Comment by czgdp1807** [comment]
Closing this. We will come to it in future. As of now, we are ignoring external memory algorithms.

**Comment by Arvind-raj06** [comment]
Yeah sure we can take reference from it later

**Pull request closed without merging**